client/src/modules/stock/exit/exit.js

```diff
@@ -412,19 +469,36 @@ function StockExitController(
     if (!vm.movement.entity.uuid && vm.movement.entity.type !== 'loss') {
       return Notify.danger('ERRORS.ER_NO_STOCK_DESTINATION');
     }
+
+    if (hasDuplicatedLots()) {
+      // notify on the concerned row
+      errorLineHighlight();
+      return Notify.danger('ERRORS.ER_DUPLICATED_LOT', 20000);
+    }
+
     vm.$loading = true;
     return mapExit[vm.movement.exit_type].submit()
       .then(toggleLoadingIndicator)
       .catch(Notify.handleError)
       .finally(() => reinit(form));
   }
 
+  function errorLineHighlight() {
+    // set and unset error flag for allowing to highlight again the row
+    // when the user click again on the submit button
+    vm.stockForm.store.data[vm.doublonDetectedLine][vm.ROW_ERROR_FLAG] = true;
```

Comment:
`vm.doubleDetectedLine` is a global variable and if we ever change the name, we will need to update this function twice!

In general, [global variables are](https://stackoverflow.com/questions/1841916/how-to-avoid-global-variables-in-javascript) [considered](http://wiki.c2.com/?GlobalVariablesAreBad) [bad practice](https://gist.github.com/hallettj/64478).

You can avoid using the global variable here by just passing it in as a function parameter:
```js
function errorLineHighlight(rowIdx) {
  const row = vm.stockForm.store.data[rowIdx];

  row[vm.ROW_ERROR_FLAG] = true;
  $timeout(() => {
    row[vm.ROW_ERROR_FLAG] = false;
   }, 1000);
}
```

Now you can just do:
```js
if (hasDuplicatedLots()) {
  errorLineHighlight(vm.doublonDetectedLine);
  // etc..
}
```